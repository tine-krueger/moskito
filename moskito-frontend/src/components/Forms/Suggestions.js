import styled from 'styled-components/macro'

export default function Suggestions({suggestions, onClick}) {
    return (
        <ListStyled>{suggestions.map((suggestion, index) =>  <ListElement key={index} onClick={() => onClick(suggestion)}>{suggestion.label}</ListElement>)}</ListStyled>
    )
}
const ListStyled = styled.ul`
    margin-top: -1.7em;
    padding: 0.5em 1em;
    list-style: none;
    padding-left: 0;
    background: #d8e6e4;
    opacity: .8;
    border-radius: 10px;
    box-shadow: inset 2px 1px 6px 0 rgba(0,0,0,0.2), inset -3px -4px 6px 0 rgba(255,255,255,.4);
   
`
const ListElement = styled.li`
    padding: .5em 1em;

    &:hover,
    &:focus,
    &:active {
        color: #d3a392;
    }
`
