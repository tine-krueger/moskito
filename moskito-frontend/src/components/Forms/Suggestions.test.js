import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import 'jest-styled-components'
import Suggestions from './Suggestions'

const suggestions= [
    {label: 'Berlin'}, 
    {label: 'Hamburg'},
    {label: 'Wanne-Eikel'}
]
const onClick = jest.fn()

describe('Suggestions', () => {

    it('renders correctly', () => {
        const { container } = render(<Suggestions suggestions={suggestions} onClick={onClick}/>)
        expect(container).toMatchSnapshot()
    })

    it('shows the citys as suggestions', () => {
        const { getByText } = render(<Suggestions suggestions={suggestions} onClick={onClick}/>)
        expect(getByText('Berlin')).toBeInTheDocument()
        expect(getByText('Hamburg')).toBeInTheDocument()
        expect(getByText('Wanne-Eikel')).toBeInTheDocument()
    })

    it('by onClick it calls function with the right city', () => {
        const { getByText } = render(<Suggestions suggestions={suggestions} onClick={onClick}/>)
        userEvent.click(getByText('Berlin'))
        expect(onClick).toBeCalledWith({label: "Berlin"})
        expect(onClick).not.toBeCalledWith({label: "Wanne-Eikel"})
    })

})