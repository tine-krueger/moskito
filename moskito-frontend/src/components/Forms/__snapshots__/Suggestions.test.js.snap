// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions renders correctly 1`] = `
.c0 {
  margin-top: -1em;
  padding: 0.5em 1em;
  list-style: none;
  padding-left: 0;
  background: #d8e6e4;
  opacity: .8;
  border-radius: 10px;
  box-shadow: inset 2px 1px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,.4);
}

.c1 {
  padding: .5em 1em;
}

.c1:hover,
.c1:focus,
.c1:active {
  color: #d3a392;
}

<div>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      Berlin
    </li>
    <li
      class="c1"
    >
      Hamburg
    </li>
    <li
      class="c1"
    >
      Wanne-Eikel
    </li>
  </ul>
</div>
`;
